test/libsolidity/syntaxTests/inlineAssembly/invalid/external_function_pointer_selector_assignment.sol

```diff
@@ -0,0 +1,17 @@
+contract C {
+	function testFunction() external {}
+
+	function testYul() public {
+		function() external fp = this.testFunction;
+
+		assembly {
+			fp.selector := 0x12345678
+		}
+	}
+	function testSol() public {
+		this.testFunction.selector = bytes4(bytes32(uint256(0x1234568)));
```

Comment:
So apparently we allow an explicit converstion from `uint32` to `bytes4` but not from an integer literal that fits in `uint32` to `bytes4`. @chriseth @hrkrshnn is that intentional?

Reply:
Yes, intentional - doing `bytes4(7)` would perform two steps at a time in my opinion.